Debounce/IBufferEvents.cs

```diff
@@ -0,0 +1,28 @@
+﻿// SPDX-FileCopyrightText: 2024 Alain van den Berg
+//
+// SPDX-License-Identifier: MIT
+
+namespace Dorssel.Utilities;
+
+/// <summary>
+/// This interface specifies the public API for the <see cref="Bufferer{TEvent}"/> class.
+/// </summary>
+public interface IBufferEvents<TEvent> : IDebounceSettings
+{
+    /// <summary>
+    /// This event will be sent when <see cref="Trigger"/> has been called one or more times and
+    /// the debounce timer times out and will contain all events that have been triggered.
+    /// </summary>
+    public event EventHandler<BufferedEventArgs<TEvent>> Buffered;
+
+    /// <summary>
+    /// Trigger an event of type <typeparamref name="TEvent"/>.
+    /// </summary>
+    /// <param name="evt">Event to trigger.</param>
+    public void Trigger(TEvent evt);
```

Comment:
Unfortunately, this one is the *real* reason why we cannot have a common interface.
It is in proposal, but already for 6+ years: https://github.com/dotnet/csharplang/discussions/696

On the other hand ... I could live with the debouncer taking an unused parameter. (let's say "int"), so we *could* have a common interface after all, where we separate the generic parameters into TEventArgs and TEvent.

For the debouncer, TEvent = int (a dummy, should really be void), TEventArgs = DebounceEventArgs.
For the bufferer, TEvent = user provided, TEventArgs = BufferedEventArgs<TEvent>

Reply:
I guess so.